lib/spack/spack/util/spack_yaml.py

```diff
@@ -153,59 +156,89 @@ def construct_yaml_map(self, node):
 
 
 # register above new constructors
-OrderedLineLoader.add_constructor("tag:yaml.org,2002:map", OrderedLineLoader.construct_yaml_map)
-OrderedLineLoader.add_constructor("tag:yaml.org,2002:seq", OrderedLineLoader.construct_yaml_seq)
-OrderedLineLoader.add_constructor("tag:yaml.org,2002:str", OrderedLineLoader.construct_yaml_str)
+OrderedLineConstructor.add_constructor(
+    "tag:yaml.org,2002:map", OrderedLineConstructor.construct_yaml_map
+)
+OrderedLineConstructor.add_constructor(
+    "tag:yaml.org,2002:seq", OrderedLineConstructor.construct_yaml_seq
+)
+OrderedLineConstructor.add_constructor(
+    "tag:yaml.org,2002:str", OrderedLineConstructor.construct_yaml_str
+)
 
 
-class OrderedLineDumper(RoundTripDumper):
-    """Dumper that preserves ordering and formats ``syaml_*`` objects.
+class OrderedLineRepresenter(representer.RoundTripRepresenter):
+    """Representer that preserves ordering and formats ``syaml_*`` objects.
 
-    This dumper preserves insertion ordering ``syaml_dict`` objects
+    This representer preserves insertion ordering ``syaml_dict`` objects
     when they're written out.  It also has some custom formatters
     for ``syaml_*`` objects so that they are formatted like their
     regular Python equivalents, instead of ugly YAML pyobjects.
-
     """
 
     def ignore_aliases(self, _data):
         """Make the dumper NEVER print YAML aliases."""
         return True
 
     def represent_data(self, data):
-        result = super(OrderedLineDumper, self).represent_data(data)
+        result = super().represent_data(data)
         if data is None:
             result.value = syaml_str("null")
         return result
 
     def represent_str(self, data):
         if hasattr(data, "override") and data.override:
             data = data + ":"
-        return super(OrderedLineDumper, self).represent_str(data)
+        return super(OrderedLineRepresenter, self).represent_str(data)
```

Comment:
Any particular reason you used `super()` in other places and the older form here?

Reply:
Fixed in e5517d5